### ISSUE TITLE
Add telemetry for studies

### ISSUE THREAD

**Issue by Amejia481** [mozilla-mobile/fenix#22192]
https://mozilla-hub.atlassian.net/browse/FNXV2-18195

## Description & Product Manager / Data Scientist User Story

- Toggling preference on/off - Event
- Indication of whether user has studies enabled - metric, boolean

## Acceptance Criteria
- [ ] ENG files a [DS JIRA](https://jira.mozilla.com/projects/DO/issues/DO-228?filter=allopenissues) request outlining their methodology.
- [ ] DS sign off on instrumentation methodology addressing product questions.
- [ ] Event pings can be queried via re:dash
- [ ] Event pings can be queried via amplitude
- [ ] We are sending telemetry events for the actions listed in the requirements
- [ ] We have documented the telemetry
- [ ] We have asked a data steward to [review](https://github.com/mozilla/data-review/blob/master/request.md) the telemetry

**Comment by Amejia481** [comment]
QA team could you help us to verify:

Every time the user goes to Settings ->  Data collection ->  Studies -> when switching the toggle we send the event
category `preferences`.`studies_preference_enabled` and on startup the metric  `preferences`.`studies_enabled`

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on Nightly 95.0a1 from 11/02 using a Google Pixel 3a (Android 11)

✅   Disable "Studies"  - Ping 80daf1f5-b03e-473e-b6b3-8c3c90521343
```
{
          "timestamp": 1109,
          "category": "preferences",
          "name": "studies_preference_enabled"
        }
```

✅   Metrics Ping - d5dea1d8-bda3-4d4e-9fa9-5f7d1ca4feab
`"preferences.studies_enabled": false`

✅   Disable "Studies"  - Ping 96e45388-52fa-49f4-8095-79961bb4dbaf
```
        {
          "timestamp": 86124,
          "category": "preferences",
          "name": "studies_preference_enabled"
        }
```

✅   Metrics Ping - f55175e6-893f-4e10-a59e-0bb473f7ab31
`"preferences.studies_enabled": true`

[Logcat](https://github.com/mozilla-mobile/fenix/files/7459369/studies.txt)
[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/AJ-02110)

**Comment by Amejia481** [comment]
Thanks @AndiAJ :)
